observation/result_writer_test.go

```diff
@@ -1,38 +1,49 @@
 package observation_test
 
 import (
+	"context"
+	"testing"
+
+	kafka "github.com/ONSdigital/dp-kafka"
+	"github.com/ONSdigital/dp-kafka/kafkatest"
 	"github.com/ONSdigital/dp-observation-importer/observation"
 	"github.com/ONSdigital/dp-observation-importer/schema"
-	"github.com/ONSdigital/go-ns/kafka/kafkatest"
 	. "github.com/smartystreets/goconvey/convey"
-	"testing"
 )
 
 func TestResultWriter_Write(t *testing.T) {
+	ctx := context.Background()
 
 	Convey("Given an observation inserted event", t, func() {
-
 		expectedInstanceID := "123abc"
 		result := &observation.Result{ObservationsInserted: 2, InstanceID: expectedInstanceID}
 		expectedEvent := observation.InsertedEvent{ObservationsInserted: 2, InstanceID: expectedInstanceID}
 
-		// mock schema producer contains the output channel to capture messages sent.
-		outputChannel := make(chan []byte, 1)
-		mockMessageProducer := kafkatest.NewMessageProducer(outputChannel, nil, nil)
-
+		pChannels := kafka.CreateProducerChannels()
+		mockMessageProducer := kafkatest.NewMessageProducerWithChannels(pChannels, false)
 		observationMessageWriter := observation.NewResultWriter(mockMessageProducer)
 
 		Convey("When write is called on the result writer", func() {
 
-			observationMessageWriter.Write([]*observation.Result{result})
-
-			Convey("The schema producer has the observation on its output channel", func() {
-
-				messageBytes := <-outputChannel
-				close(outputChannel)
-				observationEvent := Unmarshal(messageBytes)
-				So(observationEvent.InstanceID, ShouldEqual, expectedEvent.InstanceID)
-				So(observationEvent.ObservationsInserted, ShouldEqual, expectedEvent.ObservationsInserted)
+			Convey("The schema producer has the observation on its output channel", func(c C) {
+				go func() {
+					messageBytes := <-pChannels.Output
+
+					observationEvent, err := Unmarshal(messageBytes)
+					c.So(err, ShouldBeNil)
+					c.So(observationEvent.InstanceID, ShouldEqual, expectedEvent.InstanceID)
+					c.So(observationEvent.ObservationsInserted, ShouldEqual, expectedEvent.ObservationsInserted)
+
+				}()
+				observationMessageWriter.Write(ctx, []*observation.Result{result})
+
+				// Closing channels, kafkatest producer channels should contain close function
```

Comment:
This has now been implemented in the latest version of dp-kafka. Instead of manually closing the channels, you can call:
```
mockMessageProducer.Close(ctx)
```